### ISSUE TITLE
Issues with JK-B1A24S

### ISSUE THREAD

**Issue by neoclubber** [fl4p/batmon-ha#170]
I have issue with v3 hw, v3.0.7 sw JK-B1A24S. Batmon runs as add-on under hass os. Seems that batmon can't parse data from this bms correctly. It shows something like 1000000V, 700C temp etc. The app nicely reads it! Thanks!

**Comment by neoclubber** [comment]
UPD. Product page https://www.jkbms.com/product/jk-b1a24s/

**Comment by neoclubber** [comment]
Seems that v3 is JK-04 protocol. https://github.com/syssi/esphome-jk-bms/discussions/48

**Comment by fl4p** [comment]
You're right — JK-B1A24S (hw v3, fw 3.x) uses the **JK04** wire format, which batmon-ha does not implement. The current decoder in `bmslib/models/jikong.py` only handles **JK02_24S** (fw <11) and **JK02_32S** (fw ≥11) — both `55 AA EB 90 …` framed with integer cell millivolts. JK04 frames cell voltages as float32 at different offsets, so pointing the JK02 decoder at a JK04 stream produces exactly the garbage you saw (1,000,000 V, 700 °C).

This gap is noted in the README ("Implement old JK04?") and `docs/BMS_FACTSHEET.md`. The upstream aiobmsble library doesn't have a JK04 decoder either, so there's no quick adapter route.

Closing as not supported / not planned for now. Reference implementation if anyone wants to PR it: https://github.com/syssi/esphome-jk-bms/blob/main/components/jk_bms_ble/jk_bms_ble.cpp (JK04 branch).
